src/components/ConditionalFilter/FiltersQueryBuilder/queryVarsBuilders/MetadataAdvancedFilterQueryVarsBuilder.ts

```diff
@@ -0,0 +1,75 @@
+import { Handler, NoopValuesHandler } from "../../API/Handler";
+import { FilterElement } from "../../FilterElement";
+import { isTuple } from "../../FilterElement/ConditionValue";
+import { WhereOnlyQueryVarsBuilder } from "./types";
+
+/** Builds query for MetadataFilterInput fields
+ * This is different field from MetadataInput, for these use MetadataFilterQueryVarsBuilder
+ *
+ * It supports making a query for single `key` with multiple `values`
+ * E.g. {metadata: {key: "key", value: {oneOf: ["a", "b"]}}}
+ * if user created multiple input entries: key - a; key - b
+ * */
+export class MetadataAdvancedFilterQueryVarsBuilder
```

Comment:
Rename: `MetadataFilterInputQueryVarsBuilder`

because we build for `MetadataFilterInput` type in Saleor, which doesn't accept array